src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,163 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+
+    private User firstUser;
+    private User secondUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        firstUser = new User();
+        firstUser.setId(133L);
+        firstUser.setAge(22);
+        firstUser.setLogin("login");
+        firstUser.setPassword("password");
+        secondUser = new User();
+        secondUser.setId(133L);
+        secondUser.setAge(22);
+        secondUser.setLogin("secondLogin");
+        secondUser.setPassword("password");
+    }
+
+    @Test
+    void register_validUserOlderThen18years_ok() {
+        User registeredUser = registrationService.register(firstUser);
+        User userFromStorage = storageDao.get("login");
+        assertEquals(firstUser, registeredUser,
+                "User returned from register method not equals the registered user");
+        assertEquals(firstUser, userFromStorage, "Can't find registered user in the storage");
+    }
+
+    @Test
+    void register_validUser18years_ok() {
+        firstUser.setLogin("user18years");
+        firstUser.setAge(18);
+        User registeredUser = registrationService.register(firstUser);
+        User userFromStorage = storageDao.get("user18years");
+        assertEquals(firstUser, registeredUser,
+                "User returned from register method not equals the registered user");
+        assertEquals(firstUser, userFromStorage, "Can't find registered user in the storage");
+    }
+
+    @Test
+    void register_validUserPasswordSixLetters_ok() {
```

Comment:
```suggestion
    void register_validUserPassword_ok() {
```
dont put concrete numbers in test method names